src/gstnlaunch.c

```diff
@@ -541,11 +610,12 @@ main (int argc, char **argv)
     {"interactive", 'i', 0, G_OPTION_ARG_NONE, &interactive,
         ("Put on interactive mode with branches in GST_STATE_READY"), NULL}
     ,
+    {"drain", 'd', 0, G_OPTION_ARG_NONE, &thiz->drain,
+        ("Send drain query on EOS to the branch's sink"), NULL}
```

Comment:
As I understand the code, the sink is drained **before** handling `EOS`, right? Maybe make that clearer in the doc.

Reply:
Aren't elements going to drain themselves on `EOS` anyway? Not sure what's the right behavior here.

Reply:
Or if you want to drain **after** EOS has been handled you could watch for the `GST_MESSAGE_EOS` message instead.

Reply:
The idea was to provide a simple tool to send this drain query, i thought it was interesting to do it in this tool. https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/1028#note_444728

Reply:
Its a bit relevant if you want the sink to consume all the pending data before exiting everything. I think it can be useful.